src/vellum/workflows/nodes/displayable/bases/inline_prompt_node/node.py

```diff
@@ -59,6 +60,32 @@ class BaseInlinePromptNode(BasePromptNode[StateType], Generic[StateType]):
     class Trigger(BasePromptNode.Trigger):
         merge_behavior = MergeBehavior.AWAIT_ANY
 
+    def _extract_required_input_variables(self, blocks: Union[List[PromptBlock], List[RichTextChildBlock]]) -> Set[str]:
+        required_variables = set()
+
+        for block in blocks:
+            if block.block_type == "VARIABLE":
+                if hasattr(block, "input_variable"):
+                    required_variables.add(block.input_variable)
+            elif block.block_type == "CHAT_MESSAGE" and block.blocks:
+                required_variables.update(self._extract_required_input_variables(block.blocks))
+            elif block.block_type == "RICH_TEXT" and block.blocks:
+                required_variables.update(self._extract_required_input_variables(block.blocks))
+
+        return required_variables
+
+    def _validate(self) -> None:
+        required_variables = self._extract_required_input_variables(self.blocks)
+        provided_variables = set(self.prompt_inputs.keys() if self.prompt_inputs else set())
+
+        missing_variables = required_variables - provided_variables
+        if missing_variables:
+            missing_vars_str = ", ".join(f"'{var}'" for var in missing_variables)
+            raise NodeException(
+                message=f"Missing required input variables: {missing_vars_str}",
```

Comment:
[nit] I'd make this message specific to block. Something like:
`f"Missing inputs referenced by `VariableBlock`s: {missing_vars_str}"